src/Invictus.Testing/LogicAppsProvider.cs

```diff
@@ -0,0 +1,307 @@
+        /// </summary>
+        /// <param name="timeout">The period to retrieve logic app runs.</param>
+        public LogicAppsProvider WithTimeout(TimeSpan timeout)
+        {
+            _timeout = timeout;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the correlation ID as client tracking when retrieving logic app runs.
+        /// </summary>
+        /// <param name="correlationId">The client tracking of the logic app runs.</param>
+        public LogicAppsProvider WithCorrelationId(string correlationId)
+        {
+            Guard.NotNull(correlationId, nameof(correlationId));
+
+            _hasCorrelationId = true;
+            _correlationId = correlationId;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the tracking property as filter when retrieving logic app runs.
+        /// </summary>
+        /// <param name="trackedPropertyName">The name of the tracked property to filter on.</param>
+        /// <param name="trackedPropertyValue">The value of the tracked property to filter on.</param>
+        public LogicAppsProvider WithTrackedProperty(string trackedPropertyName, string trackedPropertyValue)
+        {
+            Guard.NotNull(trackedPropertyName, nameof(trackedPropertyName));
+            Guard.NotNull(trackedPropertyValue, nameof(trackedPropertyValue));
+
+            _hasTrackedProperty = true;
+            _trackedPropertyName = trackedPropertyName;
+            _trackedPropertyValue = trackedPropertyValue;
+            return this;
+        }
+
+        /// <summary>
+        /// Start polling for a single logic app run corresponding to the previously set filtering criteria.
+        /// </summary>
+        public async Task<LogicAppRun> PollForSingleLogicAppRunAsync()
+        {
+            IEnumerable<LogicAppRun> logicAppRuns = await PollForLogicAppRunsAsync(numberOfItems: 1);
+            return logicAppRuns.FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Starts polling for a series of logic app runs corresponding to the previously set filtering criteria.
+        /// </summary>
+        public async Task<IEnumerable<LogicAppRun>> PollForLogicAppRunsAsync()
+        {
+            return await PollForLogicAppRunsAsync(numberOfItems: -1);
+        }
+
+        /// <summary>
+        /// Starts polling for a <paramref name="numberOfItems"/> corresponding to the previously set filtering criteria.
+        /// </summary>
+        /// <param name="numberOfItems">The minimum amount of logic app runs to retrieve.</param>
+        public async Task<IEnumerable<LogicAppRun>> PollForLogicAppRunsAsync(int numberOfItems)
+        {
+            RetryPolicy<IEnumerable<LogicAppRun>> retryPolicy =
+                Policy.HandleResult<IEnumerable<LogicAppRun>>(
+                          runs => numberOfItems <= 0 ? !runs.Any() : runs.Count() < numberOfItems)
```

Comment:
Passing `numberOfItems: -1` for all runs feels odd, what if we make `numberOfItems` nullable and look for all of them if it's not specified?

Maybe I'm nitpicking but feels less magical?

Reply:
Yes, I'll change this. Wasn't really fond of it myself. Thanks.